marketplace/commercetools-sku/src/index.js

```diff
@@ -0,0 +1,176 @@
+import { createRequestBuilder } from '@commercetools/api-request-builder';
+import { createClient } from '@commercetools/sdk-client';
+import { createAuthMiddlewareForClientCredentialsFlow } from '@commercetools/sdk-middleware-auth';
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http';
+import { createQueueMiddleware } from '@commercetools/sdk-middleware-queue';
+
+import { setup, renderSkuPicker } from 'shared-sku-app';
+
+import logo from './logo.svg';
+import descriptor from '../extension.json';
+import { dataTransformer } from './dataTransformer';
+
+function makeCTA(fieldType) {
+  return fieldType === 'Array' ? 'Select products' : 'Select a product';
+}
+
+function makeCommerceToolsClient({
+  parameters: {
+    installation: { apiEndpoint, authApiEndpoint, projectKey, clientId, clientSecret }
+  }
+}) {
+  const authMiddleware = createAuthMiddlewareForClientCredentialsFlow({
+    host: authApiEndpoint,
+    projectKey: projectKey,
+    credentials: {
+      clientId: clientId,
+      clientSecret: clientSecret
+    }
+  });
+
+  const httpMiddleware = createHttpMiddleware({
+    host: apiEndpoint
+  });
+
+  const queueMiddleware = createQueueMiddleware({
+    concurrency: 5
+  });
+
+  return createClient({
+    middlewares: [authMiddleware, httpMiddleware, queueMiddleware]
+  });
+}
+
+const fetchProductPreviews = async function fetchProductPreviews(skus, config) {
+  if (!skus.length) {
+    return [];
+  }
+
+  const client = makeCommerceToolsClient({ parameters: { installation: config } });
+  const requestBuilder = createRequestBuilder({ projectKey: config.projectKey });
+  const uri = requestBuilder.productProjectionsSearch
+    .parse({
+      filter: [`variants.sku:${skus.map(sku => `"${sku}"`).join(',')}`]
+    })
+    .build();
+  const response = await client.execute({ uri, method: 'GET' });
+  if (response.statusCode === 200) {
+    const products = response.body.results.map(dataTransformer(config));
+    return products;
+  }
+
+  return dataTransformer({}); // return empty product
```

Comment:
maybe a comment why?

Reply:
good catch this was a leftover before i had refactored the method, it makes more sense to throw (then the sdk.notifier catches it and renders an error message)